services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/RouterServiceImpl.java

```diff
@@ -0,0 +1,316 @@
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Service
+public class RouterServiceImpl implements RouterService {
+
+    private Logger logger = LoggerFactory.getLogger();
+
+    @Autowired
+    private RouterDatabaseService routerDatabaseService;
+
+    @Autowired
+    private VpcRouterToVpcService vpcRouterToVpcService;
+
+    @Autowired
+    private RouteTableDatabaseService routeTableDatabaseService;
+
+    @Autowired
+    private RouteEntryDatabaseService routeEntryDatabaseService;
+
+
+    @Override
+    public Router getOrCreateVpcRouter(String projectId, String vpcId) throws CanNotFindVpc, DatabasePersistenceException, CacheException, ExistMultipleVpcRouter {
+        Router router = null;
+
+        // If VPC already has a router, return the router state
+        Map<String, Router> routerMap = null;
+        Map<String, Object[]> queryParams = new HashMap<>();
+        Object[] values = new Object[1];
+        values[0] = vpcId;
+        queryParams.put("owner", values);
+
+        routerMap = this.routerDatabaseService.getAllRouters(queryParams);
+
+        if (routerMap == null) {
+            routerMap = new HashMap<>();
+        }
+
+        if (routerMap.size() > 1) {
+            throw new ExistMultipleVpcRouter();
+        } else if (routerMap.size() == 1) {
+            for (Map.Entry<String, Router> entry : routerMap.entrySet()) {
+                router = (Router)entry.getValue();
+                return router;
+            }
+        } else {
+            // get vpc entity to create default route table and route route rule
+            VpcWebJson vpcResponse = this.vpcRouterToVpcService.getVpcWebJson(projectId, vpcId);
+            VpcEntity vpcEntity = vpcResponse.getNetwork();
+
+
+            // If VPC doesn’t have a router, create a new router, create a VPC routing table and pump-in the VPC default routing rules
+            router = createDefaultVpcRouter(projectId, vpcEntity);
+        }
+
+        return router;
+    }
+
+    @Override
+    public Router createDefaultVpcRouter(String projectId, VpcEntity vpcEntity) throws DatabasePersistenceException {
+        String routerId = UUID.randomUUID().toString();
+        String routeTableId = UUID.randomUUID().toString();
+        String routeEntryId = UUID.randomUUID().toString();
+        String owner = vpcEntity.getId();
+        String destination = vpcEntity.getCidr();
+        List<RouteTable> vpcRouteTables = new ArrayList<>();
+        List<String> ports = new ArrayList<>();
+        List<RouteEntry> routeEntities = new ArrayList<>();
+
+        // create a VPC routing table and pump-in the VPC default routing rules
+        RouteEntry routeEntry = new RouteEntry(projectId, routeEntryId, "default_vpc_routeEntry", "", destination, RouteConstant.DEFAULT_TARGET, RouteConstant.DEFAULT_PRIORITY, routeTableId, null);
+        routeEntities.add(routeEntry);
+        this.routeEntryDatabaseService.addRouteEntry(routeEntry);
```

Comment:
What's the `DEFAULT_TARGET` value here? 
For the VPC local network, such as 10.0.0.0/16, it's TARGET should be 'local' (same as AWS).

Reply:
We may need to define several type for TARGET, such as 'local', 'internet_gw', 'nat_gw', etc..., similar to AWS.

Reply:
DEFAULT_TARGET = "Local"

Reply:
okay ~ I will create the Target Enum